src/engine/QueryPlanner.cpp

```diff
@@ -1923,6 +1932,9 @@ auto QueryPlanner::createSpatialJoin(
   ColumnIndex ind = aIsSpatialJoin ? jcs[0][1] : jcs[0][0];
   const Variable& var =
       otherSubtreePlan._qet->getVariableAndInfoByColumnIndex(ind).first;
+  std::cout << spatialJoin->onlyForTestingGetTask().first << " "
+            << spatialJoin->onlyForTestingGetTask().second << " Add variable "
+            << var.name() << "\n";
```

Comment:
I found the bug, please for this PR remove the logging statements, I will write you an Email with the details.